entity-server-api/src/main/java/org/terracotta/entity/PassiveServerEntity.java

```diff
@@ -16,46 +16,56 @@
   /**
-   * Invoke a call on the given entity.  Note that passive entities can't return data to the client.
-   * This method is called both in the cases of normal client->server method invocation but also in the case of
-   * server->server passive synchronization.
+   * <p>Invoke a call on the given entity.  Note that passive entities can't return data to the client.</p>
+   * <p>This method is called both in the cases of normal client->server method invocation but also in the case of
+   *  server->server passive synchronization.</p>
+   * <p>Note that the thread used to make this call is determined by consulting the entity's ConcurrencyStrategy so it may be
+   *  called concurrently with other invokes.</p>
+   * <p>A note about passive synchronization messages:  When the active is given the opportunity to synchronize the data
+   *  under a given key, the messages it creates and sends for synchronization will be executed in that key, on the passive,
+   *  via this invoke() entry-point.</p>
    *
    * @param message The message from a client or upstream active server
    */
   void invoke(M message);
 
   /**
-   * Called on MANAGEMENT_KEY to notify the receiver that it is about to start receiving synchronization messages.
+   * <p>Called on {@link ConcurrencyStrategy#MANAGEMENT_KEY} to notify the receiver that it is about to start receiving
+   *  synchronization messages.</p>
+   * <p>Note that this is called after createNew().</p>
    */
   void startSyncEntity();
 
   /**
-   * Called on MANAGEMENT_KEY to notify the receiver synchronization is complete and it will see no more synchronization
-   * messages.
+   * Called on {@link ConcurrencyStrategy#MANAGEMENT_KEY} to notify the receiver synchronization is complete and it will see
+   *  no more synchronization messages.
    */
   void endSyncEntity();
 
   /**
-   * Called on concurrencyKey to notify the receiver that it is about to start receiving synchronization messages on this
-   * key.
+   * <p>Called on concurrencyKey to notify the receiver that it is about to start receiving synchronization messages on this
+   *  key.</p>
+   * <p>To clarify:  the concurrencyKey is both the argument (so the entity can logically act on it) but is also the key
+   *  where this call is executed.</p>
    */
   void startSyncConcurrencyKey(int concurrencyKey);
 
   /**
-   * Called on concurrencyKey to notify the receiver that synchronization of this key is complete and no more
-   * synchronization messages will be received on that key.
+   * <p>Called on concurrencyKey to notify the receiver that synchronization of this key is complete and no more
+   *  synchronization messages will be received on that key.</p>
+   * <p>To clarify:  the concurrencyKey is both the argument (so the entity can logically act on it) but is also the key
+   *  where this call is executed.</p>
    */
   void endSyncConcurrencyKey(int concurrencyKey);
```

Comment:
Shouldn't we add a note here that `invoke` calls on a concurrencyKey will start only after this method has been completed for that same key? And that `invoke` calls on a concurrencyKey does not wait till entire sync is complete (`endSyncEntity(...)`) but can start immediately after the `endSyncConcurrencyKey(...)` invocation is complete for that key?